zipline/testing/predicates.py

```diff
@@ -552,6 +552,13 @@ def assert_array_equal(result,
         if array_decimal is None else
         partial(np.testing.assert_array_almost_equal, decimal=array_decimal)
     )
+
+    is_datetime = False
+    if result.dtype.kind == 'M' and expected.dtype.kind == 'M':
```

Comment:
You probably also want to handle `'m'` (timedelta) the same way. I'd write that check as:

```
if result.dtype.kind in 'mM' and expected.dtype == result.dtype:
```